src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -53,10 +95,129 @@ export class RetireJsEngine implements RuleEngine {
-		return [];
+		// RetireJS doesn't accept individual files. It only accepts directories. So we need to resolve all of the files
+		// we were given into a directory that we can pass into RetireJS.
+		const tmpDir = await this.createTmpDirWithDuplicatedTargets(target);
+		const invocationArray: RetireJsInvocation[] = this.buildCliInvocations(rules, tmpDir);
+
+		const retireJsPromises: Promise<RuleResult[]>[] = [];
+		for (const invocation of invocationArray) {
+			retireJsPromises.push(this.executeRetireJs(invocation));
+		}
+
+		// We can combine the results into a single array using .reduce() instead of the more verbose for-loop.
+		return (await Promise.all(retireJsPromises)).reduce((acc, r) => [...acc, ...r], []);
+	}
+
+	private buildCliInvocations(rules: Rule[], target: string): RetireJsInvocation[] {
+		const invocationArray: RetireJsInvocation[] = [];
+		for (const rule of rules) {
+			switch (rule.name) {
+				case 'insecure-bundled-dependencies':
+					// This rule is looking for files that contain insecure libraries, e.g. .min.js or similar.
+					// So we use --js and --jspath to make retire-js only examine JS files and skip node modules.
+					invocationArray.push({
+						args: ['--js', '--jspath', target, '--outputformat', 'json'],
+						rule: rule.name
+					});
+					break;
+				default:
+					throw new SfdxError(`Unexpected retire-js rule: ${rule.name}`);
+			}
+		}
+		return invocationArray;
+	}
+
+	private async executeRetireJs(invocation: RetireJsInvocation): Promise<RuleResult[]> {
+		return new Promise<RuleResult[]>((res, rej) => {
+			const cp = childProcess.spawn(RetireJsEngine.RETIRE_JS_PATH, invocation.args);
+
+			// We only care about StdErr, since that's where the vulnerability entries will be logged to.
+			let stderr = '';
+
+			// When data is passed back up to us, pop it onto the appropriate string.
+			cp.stderr.on('data', data => {
+				stderr += data;
+			});
+
+			cp.on('exit', code => {
+				this.logger.trace(`executeRetireJs has received exit code ${code}`);
+				if (code === 0) {
+					// If RetireJS exits with code 0, then it ran successfully and found no vulnerabilities. We can resolve
+					// to an empty array.
+					res([]);
+				} else if (code === 13) {
+					// If RetireJS exits with code 13, then it ran successfully, but found at least one vulnerability.
+					// Convert the output into RuleResult objects and resolve to that.
+					res(this.processOutput(stderr, invocation.rule));
+				} else {
+					// If RetireJS exits with any other code, then it means something went wrong. We'll reject with stderr
+					// for the ease of upstream error handling.
+					rej(stderr);
+				}
+			});
+		});
+	}
+
+	protected processOutput(cmdOutput: string, ruleName: string): RuleResult[] {
+		// The output from the CLI should be a valid JSON.
+		const outputJson = JSON.parse(cmdOutput);
```

Comment:
Should we also handle the unexpected situation when cmdOutput may not be a valid Json? Would be nicer to have a meaningful SfdxError than let the parse error show up for users.

Reply:
Good catch. I'll add that.